From e793dbed26593074d1ccc20980344a3bd2820fb4 Mon Sep 17 00:00:00 2001
From: Zaikuo Wang <zaikuo.wang@intel.com>
Date: Fri, 3 Jul 2015 14:27:32 +0800
Subject: [PATCH 440/462] ov8858/dw9718:add some log to debug power issue

add some error log in error handling

Change-Id: If9ceb6ce39eb29259143bd5d65c5093e3880fc17
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-39671
Signed-off-by: Zaikuo Wang <zaikuo.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/63361
Approver: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: Yunliang Ding <yunliang.ding@intel.com>
Reviewed-by: Soren Friis <soren.friis@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/i2c/imx/dw9718.c          | 25 ++++++++++++++++------
 .../camera/drivers/media/i2c/ov8858.c              | 12 ++++++++---
 2 files changed, 27 insertions(+), 10 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.c b/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.c
index 9139a30..1cf2cb3 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/imx/dw9718.c
@@ -87,8 +87,10 @@ int dw9718_t_focus_abs(struct v4l2_subdev *sd, s32 value)
 	value = clamp(value, 0, DW9718_MAX_FOCUS_POS);
 	ret = dw9718_i2c_wr16(client, DW9718_DATA_M, value);
 	/*pr_info("%s: value = %d\n", __func__, value);*/
-	if (ret < 0)
+	if (ret < 0) {
+		dev_err(&client->dev, "write DW9718_DATA_M  failed %d\n", ret);
 		return ret;
+	}
 
 	getnstimeofday(&dw9718_dev.focus_time);
 	dw9718_dev.focus = value;
@@ -107,19 +109,24 @@ int dw9718_vcm_power_up(struct v4l2_subdev *sd)
 
 	/* Enable power */
 	ret = dw9718_dev.platform_data->power_ctrl(sd, 1);
-	if (ret)
+	if (ret) {
+		dev_err(&client->dev, "DW9718_PD power_ctrl failed %d\n", ret);
 		return ret;
+	}
 	ret = dw9718_i2c_wr8(client, DW9718_PD, 0);
-	if (ret < 0)
+	if (ret < 0) {
+		dev_err(&client->dev, "write DW9718_PD to 0 failed %d\n", ret);
 		goto fail_powerdown;
+	}
 	/* Wait for VBAT to stabilize */
 	udelay(100);
 
 	/* Detect device */
 	ret = dw9718_i2c_rd8(client, DW9718_SACT, &value);
-	if (ret < 0)
+	if (ret < 0) {
+		dev_err(&client->dev, "read DW9718_SACT failed %d\n", ret);
 		goto fail_powerdown;
-
+	}
 	/*
 	 * WORKAROUND: for module P8V12F-203 which are used on
 	 * Cherrytrail Refresh Davis Reef AoB, register SACT is not
@@ -137,13 +144,17 @@ int dw9718_vcm_power_up(struct v4l2_subdev *sd)
 			     DW9718_CONTROL_S_SAC4 |
 			     DW9718_CONTROL_OCP_DISABLE |
 			     DW9718_CONTROL_UVLO_DISABLE);
-	if (ret < 0)
+	if (ret < 0) {
+		dev_err(&client->dev, "write DW9718_CONTROL  failed %d\n", ret);
 		goto fail_powerdown;
+	}
 	ret = dw9718_i2c_wr8(client, DW9718_SACT,
 			     DW9718_SACT_MULT_TWO |
 			     DW9718_SACT_PERIOD_8_8MS);
-	if (ret < 0)
+	if (ret < 0) {
+		dev_err(&client->dev, "write DW9718_SACT  failed %d\n", ret);
 		goto fail_powerdown;
+	}
 
 	ret = dw9718_t_focus_abs(sd, dw9718_dev.focus);
 	if (ret)
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index 936005f..b9c711d 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -798,18 +798,24 @@ static int power_up(struct v4l2_subdev *sd)
 
 	/* Enable power */
 	ret = __power_ctrl(sd, 1);
-	if (ret)
+	if (ret) {
+		dev_err(&client->dev, "power rail on failed %d.\n", ret);
 		goto fail_power;
+	}
 
 	/* Enable clock */
 	ret = dev->platform_data->flisclk_ctrl(sd, 1);
-	if (ret)
+	if (ret) {
+		dev_err(&client->dev, "flisclk on failed %d\n", ret);
 		goto fail_clk;
+	}
 
 	/* Release reset */
 	ret = __gpio_ctrl(sd, 1);
-	if (ret)
+	if (ret) {
+		dev_err(&client->dev, "gpio on failed %d\n", ret);
 		goto fail_gpio;
+	}
 
 	/* Minumum delay is 8192 clock cycles before first i2c transaction,
 	 * which is 1.37 ms at the lowest allowed clock rate 6 MHz */
-- 
1.9.1

